### PR TITLE
Readme improvements

### PR DESCRIPTION
**Pull request by ericholguin**
Minor improvements matching what is displayed in the project site, made accessible here:
- Added project image
- Links to ATC socials (Slack, Twitter, Youtube videos)
- Release badge
- Additional GHA badges with links to their respecting result history
- Links to TO API and TC modules

<hr/>

## Which Traffic Control components are affected by this PR?

None

## What is the best way to verify this PR?

Verify Readme looks good and links work properly

## PR submission checklist
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

**Review by ocket8888 — CHANGES_REQUESTED** [review]
It may be worth noting somewhere that all of the docs links point to "latest" which is the unreleased master branch, and are neither stable nor necessarily accurate for any given supported release.

File: README.md [hunk 1]
```diff
@@ -17,24 +17,58 @@
...
-![Build Status](https://github.com/apache/trafficcontrol/workflows/CDN-in-a-Box%20CI/badge.svg)
-![Weasel License Checks](https://github.com/apache/trafficcontrol/workflows/Weasel%20License%20checks/badge.svg)
-![Traffic Ops Integration Tests](https://github.com/apache/trafficcontrol/workflows/Traffic%20Ops%20Go%20client/API%20integration%20tests/badge.svg)
-![Go Unit Tests](https://github.com/apache/trafficcontrol/workflows/Go%20Unit%20Tests/badge.svg)
-![Go Formatting](https://github.com/apache/trafficcontrol/workflows/Go%20Format/badge.svg)
-![TP Integration Tests](https://github.com/apache/trafficcontrol/workflows/TP%20Integration%20Tests/badge.svg)
-![TM Integration Tests](https://github.com/apache/trafficcontrol/workflows/TM%20Integration%20Tests/badge.svg)
+# Apache Traffic Control
+
+![](https://traffic-control-cdn.readthedocs.io/en/latest/_static/ATC-SVG-FULL-WHITE.svg)
```
Comment (zrhoffman) [inline README.md:22]: This is a lot of empty space. The website has [a version of the logo](https://trafficcontrol.apache.org/resources/Traffic-Control-Logo-FINAL-Black-Text.png) ([source](https://github.com/apache/trafficcontrol-website/blob/asf-site/resources/Traffic-Control-Logo-FINAL-Black-Text.png)) that includes the name of the project, too. Should that be used?

Apache Traffic Control

**Review by zrhoffman — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (ericholguin) [reply README.md:22]: Ohh, oops the logo I used only works with the GitHub Dark theme. Will update this

**Review by zrhoffman — APPROVED** [review]
Logo looks good!

**Review by ocket8888 — APPROVED** [review]

**Pull request merged**